src/model.py

```diff
@@ -151,6 +153,16 @@ def init_engine(self):
             AsyncEngineArgs(**self.vllm_engine_config)
         )
 
+        # If TRITON_ENABLE_METRICS<_CPU/_GPU> build flag is enabled.
+        if self.args["metrics_mode"] in ["all", "cpu", "gpu"]:
```

Comment:
Instead of introducing this new `metrics_mode` [here](https://github.com/triton-inference-server/python_backend/pull/375/commits/6404f14e0b62816048b6e1272a2521b1780810fc#diff-f2bb64cbc9d4dc5d938ce965813b23178f53732e02e8c3a2afa707e287e92e57R567-R583), does it make sense to do something simpler like this?

```python
class VllmStatLogger:
  def __init__(self):
    try:
      self.metrics = TritonMetrics(labels=labels)
    # Get some "metrics not supported" error if metrics disabled
    except TritonModelException as e:
      self.metrics = None    

  def log(self):
    if not self.metrics:
      return

    # otherwise metrics are enabled so log_counter()
```

CC @Tabrizian @krishung5

Reply:
Or similarly only call `llm_engine.add_logger(logger)` if `logger` initializes successfully.

Reply:
User might want to enable only CPU/GPU metrics from vLLM.

Reply:
Personally I think we should defer that fine-grained level of support which involves addition of a new `arg` to python models to further discussion, and not include it as part of this PR to try to keep the scope as simple as possible.